### PR TITLE
Update arm_instruction.c

### PR DESCRIPTION
**Pull request by bouchrabtz**

### CODE REVIEW

**Review by DoDucAnh2810 — APPROVED** [review]
It looks good. So I will leave it to Rafael to decide what to do with testing.

File: arm_instruction.c [hunk 1]
```diff
@@ -28,10 +28,120 @@ Contact: Guillaume.Huard@imag.fr
...
+        return get_bit(zncv, Z) ==0 && (get_bit(zncv, N)==get_bit(zncv, V));
+        break;
+    case 0xD:
+        return get_bit(zncv, Z) ==1 || (get_bit(zncv, N)!=get_bit(zncv, V));
+        break;
+    case 0xE:
+        return 1;
+    break;
+    }
+    return 0;
```
Comment (DoDucAnh2810) [inline arm_instruction.c:81]: Make it a default case will be better no?

File: arm_instruction.c [hunk 2]
```diff
@@ -28,10 +28,120 @@ Contact: Guillaume.Huard@imag.fr
...
+                        break;
+    				case 0b10110:
+    					break;
+    				case 0b10000:
+                        break;
+    				case 0b10100: //undefined
+    					break;
+    				default:
+    					arm_data_processing_immediate_msr(p,instruction);
+    					break;
```
Comment (CodeNachos) [inline arm_instruction.c:116]: This is a very unusual way for testing those bits values..

File: arm_instruction.c [hunk 3]
```diff
@@ -28,10 +28,120 @@ Contact: Guillaume.Huard@imag.fr
...
+            arm_load_store_multiple(p, instruction);
+        break;
+        case 0b101: 
+            arm_branch(p, instruction);
+        break;
+        case 0b110:
+            arm_coprocessor_load_store(p, instruction);
+        break;
+        case 0b111: 
+            arm_coprocessor_others_swi(p, instruction);
```
Comment (CodeNachos) [inline arm_instruction.c:136]: software interrupt bit 24 must be 1, need to test for this bit

**Review by CodeNachos — CHANGES_REQUESTED** [review]

File: arm_instruction.c (hunk 2 above)
Comment (TheAmazingSui) [reply arm_instruction.c:116]: True

Comment (TheAmazingSui) [reply arm_instruction.c:116]: Should really use default here.

File: arm_instruction.c (hunk 3 above)
Comment (TheAmazingSui) [reply arm_instruction.c:136]: Where do you find this info?

**Comment by TheAmazingSui** [comment]
Overall, the code is correct and it is clear that you've understood what you're supposed to do.
I'd be nice if it was a bit more commented though, as it is very painful to read. Also, at some occasions it is much better to use a 'default:' condition rather than using a case with an empty break inside.

File: arm_instruction.c (hunk 3 above)
Comment (CodeNachos) [reply arm_instruction.c:136]: Screenshot_20231218-155508

Comment (CodeNachos) [reply arm_instruction.c:136]: At A3-2

**Review by CodeNachos — APPROVED** [review]

**Pull request merged**